### ISSUE TITLE
imagepipeline-base-test:compileDebugJavaWithJavac FAILED - package com.facebook.infer.annotation does not exist

### ISSUE THREAD

**Issue by eldk** [facebook/fresco#2567]
Hello,

Trying to build from Fresco Git master branch with `./gradlew build` , I've got those errors (using `./gladlew build --scan`) :

```
:imagepipeline-base-test:compileDebugJavaWithJavac FAILED
/home/eldk/TESTREBASEFRESCO/FrescoMaster/fresco/imagepipeline-base-test/src/main/java/com/facebook/imagepipeline/testing/FakeClock.java:11: error: package com.facebook.infer.annotation does not exist
import com.facebook.infer.annotation.Nullsafe;
                                    ^
/home/eldk/TESTREBASEFRESCO/FrescoMaster/fresco/imagepipeline-base-test/src/main/java/com/facebook/imagepipeline/testing/FakeClock.java:16: error: cannot find symbol
@Nullsafe(Nullsafe.Mode.STRICT)
 ^
  symbol: class Nullsafe
/home/eldk/TESTREBASEFRESCO/FrescoMaster/fresco/imagepipeline-base-test/src/main/java/com/facebook/imagepipeline/testing/ScheduledQueue.java:11: error: package com.facebook.infer.annotation does not exist
import com.facebook.infer.annotation.Nullsafe;
                                    ^
/home/eldk/TESTREBASEFRESCO/FrescoMaster/fresco/imagepipeline-base-test/src/main/java/com/facebook/imagepipeline/testing/ScheduledQueue.java:14: error: cannot find symbol
@Nullsafe(Nullsafe.Mode.LOCAL)
 ^
  symbol: class Nullsafe
/home/eldk/TESTREBASEFRESCO/FrescoMaster/fresco/imagepipeline-base-test/src/main/java/com/facebook/imagepipeline/testing/TestScheduledExecutorService.java:10: error: package com.facebook.infer.annotation does not exist
import com.facebook.infer.annotation.Nullsafe;
                                    ^
/home/eldk/TESTREBASEFRESCO/FrescoMaster/fresco/imagepipeline-base-test/src/main/java/com/facebook/imagepipeline/testing/TestScheduledExecutorService.java:16: error: cannot find symbol
@Nullsafe(Nullsafe.Mode.STRICT)
 ^
  symbol: class Nullsafe
/home/eldk/TESTREBASEFRESCO/FrescoMaster/fresco/imagepipeline-base-test/src/main/java/com/facebook/imagepipeline/testing/TestNativeLoader.java:10: error: package com.facebook.infer.annotation does not exist
import com.facebook.infer.annotation.Nullsafe;
                                    ^
/home/eldk/TESTREBASEFRESCO/FrescoMaster/fresco/imagepipeline-base-test/src/main/java/com/facebook/imagepipeline/testing/TestNativeLoader.java:16: error: cannot find symbol
@Nullsafe(Nullsafe.Mode.STRICT)
 ^
  symbol: class Nullsafe
8 errors
```
So build fails.

* Fresco version: last master branch (up to commit 40343e8d7e5fd21e864ecaeb5b375edac3657f1e)
* Platform version: ubuntu 16.04

Build is OK when switching to v2.3.0 tag.

**Comment by alsutton** [comment]
Fixed in #2574

**Comment by mikejonesguy** [comment]
I'm seeing this issue with v2.4.0 (build warning only -- the build completes successfully). Has this fix been released yet? I was able to build without warning by including the infer package as `compileOnly` dependency:
`compileOnly "com.facebook.infer.annotation:infer-annotation:0.18.0"`
